src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php

```diff
@@ -821,6 +821,21 @@ public function testValidateUniquenessWithEmptyIterator($entity, $result)
         $this->assertNoViolation();
     }
 
+    public function testValidateNonEntity()
+    {
+        $constraint = new UniqueEntity([
+            'message' => 'myMessage',
+            'fields' => ['name'],
+            'em' => self::EM_NAME,
+        ]);
+
+        $this->validator->validate(null, $constraint);
+
+        $this->assertNoViolation();
+
+        $this->validator->validate('foo', $constraint);
```

Comment:
I don't think that this validation should actually pass.

Reply:
True, throw `UnexpectedTypeException`?

Reply:
That sounds about right. 🤔

Reply:
`UnexpectedValueException` is the proper exception for this.